src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,58 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class type) {
```

Comment:
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
```
using raw types (without generics) is almost always a bad idea